activerecord/test/cases/insert_all_test.rb

```diff
@@ -282,6 +283,26 @@ def test_insert_all_and_upsert_all_with_aliased_attributes
     end
   end
 
+  def test_insert_all_and_upsert_all_with_sti
+    assert_difference -> { Category.count }, 2 do
+      SpecialCategory.insert_all [{ name: "First" }, { name: "Second", type: nil }]
+    end
+
+    first, second = Category.last(2)
+    assert_equal "SpecialCategory", first.type
+    assert_nil second.type
+
+    if supports_insert_on_duplicate_update?
+      SpecialCategory.upsert_all [{ id: 103, name: "First" }, { id: 104, name: "Second", type: nil }]
```

Comment:
:thinking: Any chance to use different model with some unique column to not hardcode ids in here? Not sure if that's possible with Rails test suite setup, but I had hard time debugging similar randomly failing specs due to this.

Reply:
This is a style used in this test file (see the test above this, for example), so 🤷.

Reply:
Yeah, I have checked. IMHO It is better late than never. Anyway it is just nitpick. PR looks good. :muscle: